nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -363,19 +363,41 @@ protected Document getChild(String parentId, String name) {
         return children;
     }
 
-    protected List<String> getChildrenIds(String parentId) {
+    protected List<String> getAllChildrenIds(String parentId) {
+        // We want all children, the filter flags are null
+        boolean excludeSpecialChildren = false;
+        boolean excludeRegularChildren = false;
+        return internalGetChildrenIds(parentId, excludeSpecialChildren, excludeRegularChildren);
+    }
+
+    protected List<String> getChildrenIds(String parentId, boolean excludeSpecialChildren,
+            boolean excludeRegularChildren) {
+        return internalGetChildrenIds(parentId, excludeSpecialChildren, excludeRegularChildren);
+    }
+
+    protected List<String> internalGetChildrenIds(String parentId, boolean excludeSpecialChildren,
+            boolean excludeRegularChildren) {
         if (isOrderable(parentId)) {
             // TODO get only id and pos, not full state
             // TODO state not for update
-            List<DBSDocumentState> docStates = transaction.getChildrenStates(parentId);
+            List<DBSDocumentState> docStates;
+            if (!excludeSpecialChildren && !excludeRegularChildren) {
+                docStates = transaction.getAllChildrenStates(parentId);
+            } else {
+                docStates = transaction.getChildrenStates(parentId, excludeSpecialChildren, excludeRegularChildren);
+            }
             docStates.sort(POS_COMPARATOR);
             List<String> children = new ArrayList<>(docStates.size());
             for (DBSDocumentState docState : docStates) {
                 children.add(docState.getId());
             }
             return children;
         } else {
-            return transaction.getChildrenIds(parentId);
+            if (!excludeSpecialChildren && !excludeRegularChildren) {
+                return transaction.getAllChildrenIds(parentId);
+            } else {
+                return transaction.getChildrenIds(parentId, excludeSpecialChildren, excludeRegularChildren);
+            }
```

Comment:
The whole `if` block can now be replaced by just `return transaction.getChildrenIds(parentId, excludeSpecialChildren, excludeRegularChildren)`.